format.c

```diff
@@ -6,8 +6,17 @@
 #include <fadec.h>
 
 
+#ifdef _MSC_VER
+#define LIKELY(x) (x)
+#define UNLIKELY(x) (x)
+#define DECLARE_ARRAY_SIZE(n) n
+#define DECLARE_RESTRICTED_ARRAY_SIZE(n) n
```

Comment:
Didn't MS recently announce to fully comply with C11/C17? `restrict` and `static` inside array declarators are mandatory in C11 (and maybe even in C99).

Reply:
Not fully I'm afraid:

```
FAILED: subprojects/fadec/libfadec.a.p/format.c.obj
"cl" "-Isubprojects\fadec\libfadec.a.p" "-Isubprojects\fadec" "-I..\subprojects\fadec" "-DNDEBUG" "/MDd" "/nologo" "/showIncludes" "/utf-8" "/W4" "/std:c11" "/O1" "/Gw" "/Z7" "-D_CRT_SECURE_NO_WARNINGS" "/wd4018" "/wd4244" "/wd4245" "/wd4267" "/wd4310" /Fosubprojects/fadec/libfadec.a.p/format.c.obj "/c" ../subprojects/fadec/format.c
../subprojects/fadec/format.c(80): error C2143: syntax error: missing ']' before 'type'
../subprojects/fadec/format.c(80): error C2143: syntax error: missing ')' before 'type'
../subprojects/fadec/format.c(80): error C2143: syntax error: missing ';' before 'constant'
../subprojects/fadec/format.c(80): error C2059: syntax error: ')'
../subprojects/fadec/format.c(425): warning C4020: 'fd_strpcatnum': too many actual parameters
../subprojects/fadec/format.c(452): warning C4020: 'fd_strpcatnum': too many actual parameters
../subprojects/fadec/format.c(464): warning C4020: 'fd_strpcatnum': too many actual parameters
ninja: build stopped: subcommand failed.
```

This is with MSVC 19.33.31629, part of VS2022 17.3.0.